bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/LayerCollapseHandler.js

```diff
@@ -26,10 +26,21 @@ const getLayerGroups = (groups = []) => {
 
 const providerReducer = (accumulator, currentLayer) => {
     const id = currentLayer.getDataProviderId();
-    if (!id) {
-        return accumulator;
+    let dataProviderId;
+    if (typeof currentLayer.getId() === 'string') {
+        if (currentLayer.getId().includes('myplaces')) {
+            dataProviderId = 'myplaces';
+        } else if (currentLayer.getId().includes('analysis')) {
+            dataProviderId = 'analysis';
+        } else if (currentLayer.getId().includes('userlayer')) {
+            dataProviderId = 'userlayer';
+        }
+    } else {
+        if (!id) {
+            return accumulator;
+        }
+        dataProviderId = '' + id;
     }
```

Comment:
We can revert this with modelbuilder creating the dataprovider and setting the provider id